system-addon/test/functional/mochitest/head.js

```diff
@@ -93,3 +111,86 @@ function test_newtab(testInfo) { // eslint-disable-line no-unused-vars
   Object.defineProperty(testTask, "name", {value: contentTask.name});
   add_task(testTask);
 }
+
+async function check_highlights_elements(selector, length, message) { // eslint-disable-line no-unused-vars
+  // simulate a newtab open as a user would
+  BrowserOpenTab();
+
+  // wait until the browser loads
+  let browser = gBrowser.selectedBrowser;
+  await waitForPreloaded(browser);
+
+  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", false);
+  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", true);
+
+  await BrowserTestUtils.waitForCondition(() => content.document.querySelector(selector), "wait for element");
```

Comment:
The failure state is pretty hard to debug at the moment, this should be maybe `No elements matching ${selector} were found`